src/test/miner_tests.cpp

```diff
@@ -672,8 +677,10 @@ BOOST_AUTO_TEST_CASE(CreateNewBlock_validity)
         const int current_height{mining->getTip()->height};
 
         // Simple block creation, nothing special yet:
-        block_template = mining->createNewBlock(options);
-        BOOST_REQUIRE(block_template);
+        if (current_height % 2 == 0) {
+            block_template = mining->createNewBlock(options);
+            BOOST_REQUIRE(block_template);
+        } // for odd heights the next template is created at the end of this loop
```

Comment:
In commit "Add waitNext() to BlockTemplate interface" (7f6f58bad8e56c9a25ffbc0d105bed37e9958fef)

I found this comment pretty confusing and would suggest making it more specific like "// if current_height is odd, block_template will have already been set at the end of the previous loop."